src/main/java/org/shanerx/tradeshop/objects/Shop.java

```diff
@@ -0,0 +1,756 @@
+		s.update();
+	}
+
+	/**
+	 * Updates the text on the shops sign
+	 */
+	public void updateSign(SignChangeEvent signEvent) {
+		if (!missingItems()) {
+			signEvent.setLine(0, ChatColor.DARK_GREEN + shopType.toHeader());
+		} else {
+			signEvent.setLine(0, ChatColor.GRAY + shopType.toHeader());
+		}
+
+		if (product != null) {
+			StringBuilder sb = new StringBuilder();
+
+			sb.append(product.getAmount());
+			sb.append(" ");
+
+			sb.append((product.hasItemMeta() && product.getItemMeta().hasDisplayName()) ?
+					product.getItemMeta().getDisplayName() :
+					product.getType().toString());
+
+			signEvent.setLine(1, sb.toString().substring(0, (sb.length() < 15) ? sb.length() : 15));
+
+		} else {
+			signEvent.setLine(1, "");
+		}
+
+		if (cost != null) {
+			StringBuilder sb = new StringBuilder();
+
+			sb.append(cost.getAmount());
+			sb.append(" ");
+
+			sb.append((cost.hasItemMeta() && cost.getItemMeta().hasDisplayName()) ?
+					cost.getItemMeta().getDisplayName() :
+					cost.getType().toString());
+
+			signEvent.setLine(2, sb.toString().substring(0, (sb.length() < 15) ? sb.length() : 15));
+		} else {
+			signEvent.setLine(2, "");
+		}
+
+		signEvent.setLine(3, status.getLine());
+	}
+
+	/**
+	 * Returns the shops inventory as a BlockState
+	 *
+	 * @return shops inventory as BlockState
+	 */
+	public BlockState getInventory() {
+		try {
+			return getInventoryLocation().getBlock().getState();
+		} catch (NullPointerException npe) {
+			return null;
+		}
+	}
+
+	/**
+	 * Returns the shops status as ShopStatus
+	 *
+	 * @return Shops status as ShopStatus
+	 */
+	public ShopStatus getStatus() {
+		return status;
+	}
+
+	/**
+	 * Sets the shop to open if the shop has all necessary information to make a trade
+	 *
+	 * @return true if shop opened
+	 */
+	public boolean setOpen() {
+		boolean ret;
+
+		if (!missingItems()) {
+			status = ShopStatus.OPEN;
+			ret = true;
+		} else {
+			status = ShopStatus.CLOSED;
+			ret = false;
+		}
+
+		updateSign();
+		return ret;
+	}
+
+	/**
+	 * Checks if shop has necessary items to make a trade
+	 *
+	 * @return true if items are missing
+	 */
+	public boolean missingItems() {
+		return product == null || cost == null;
+	}
+
+	/**
+	 * Sets the shops status to closed
+	 */
+	public void setClosed() {
+		status = ShopStatus.CLOSED;
+		updateSign();
+	}
+
+	/**
+	 * Removes this shop from file
+	 */
+	public void remove() {
+		JsonConfiguration json = new JsonConfiguration(shopLoc.getLocation().getChunk());
+
+		json.removeShop(shopLoc);
+	}
+
+	/**
+	 * Checks if shop is open
+	 *
+	 * @return true if open
+	 */
+	public boolean isOpen() {
+		return status == ShopStatus.OPEN;
+	}
+
+	/**
+	 * Switches the type of the shop between 'Trade' and 'BiTrade'
+	 */
+	public void switchType() {
+		if (shopType == ShopType.TRADE)
+			setShopType(ShopType.BITRADE);
+		else if (shopType == ShopType.BITRADE)
+			setShopType(ShopType.TRADE);
+
+		updateSign();
+		saveShop();
+	}
+}
```

Comment:
Were any mechanics changed or was this just cleaned up/moved to another package?

Reply:
Besides being made for 2.0 I think the only changes recently are minor fixes.